install/components/ylab/likes/class.php

```diff
@@ -39,6 +67,177 @@ public function onPrepareComponentParams($arParams)
         return parent::onPrepareComponentParams($arParams);
     }
 
+    /**
+     * @param $sSignedParameters
+     * @return false
+     * @throws \Bitrix\Main\ArgumentException
+     * @throws \Bitrix\Main\ArgumentTypeException
+     * @throws \Bitrix\Main\LoaderException
+     * @throws \Bitrix\Main\ObjectPropertyException
+     * @throws \Bitrix\Main\Security\Sign\BadSignatureException
+     * @throws \Bitrix\Main\SystemException
+     */
+    public function setLikeAction($sSignedParameters)
+    {
+        /** @var \CUser $USER */
+        global $USER;
+
+        if(!$USER->IsAuthorized()) return false;
+
+        $signer = new ParameterSigner;
+        $this->arParams = $signer->unsignParameters($this->__name, $sSignedParameters);
+        $this->arParams = $this->onPrepareComponentParams($this->arParams);
+
+        $oResultGet = YlabLikesTable::getList([
+            'filter' => [
+                'CONTENT_ID' => $this->arParams['ELEMENT_ID'],
+                'CONTENT_TYPE' => $this->arParams['ENTITY_ID'],
+                'USER_ID' => $USER->GetID()
+            ],
+            'select' => ['ID', 'VOTE']
+        ]);
+        if ($oResultGet->getSelectedRowsCount() > 0) {
+            $arResultGet = $oResultGet->fetch();
+            if ($arResultGet['VOTE'] == self::VoteLike) {
+                $oResult = YlabLikesTable::delete($arResultGet['ID']);
+            } else {
+                $oResult = YlabLikesTable::update($arResultGet['ID'], ['VOTE' => self::VoteLike]);
```

Comment:
```suggestion
                $oResult = YlabLikesTable::update($arResultGet['ID'], ['VOTE' => YlabLikesTable::VoteLike]);
```
Константы значения лайка и дизлайка нужно оставить в модели